Tests/test_imageshow.py

```diff
@@ -62,4 +62,20 @@ def test_viewer():
 
 def test_viewers():
     for viewer in ImageShow._viewers:
-        viewer.get_command("test.jpg")
+        try:
+            viewer.get_command("test.jpg")
+        except NotImplementedError:
+            pass
```

Comment:
```suggestion
            assert viewer.show(hopper()) == 1
```
Would remove the need for a separate test.

Reply:
The separate test
- confirms that the IPythonViewer is present when IPython is available
- runs IPythonViewer even if another Viewer instance has priority

Also, IPython seems to be a special case, where running its Viewer doesn't disrupt the test process - thinking generically for the sake of the future, just because a Viewer isn't based on a command line instruction doesn't mean that we want it to run in the middle of the test suite?